tests/python/contrib/test_hexagon/test_cache_read_write.py

```diff
@@ -1,226 +0,0 @@
-
-
-@tvm.testing.requires_hexagon
-def test_cache_read_write_2d(hexagon_session: Session):
-    """Test 2D cache_read and cache_write to global.vtcm for hexagon"""
-    size = 128
-    outer_shape = (size,)
-    factor = 16
-    inner_shape = (factor,)
-    dtype = "int8"
-
-    x_tensor = te.placeholder(shape=outer_shape, dtype=dtype, name="x")
-    y_tensor = te.placeholder(shape=outer_shape, dtype=dtype, name="y")
-    z_tensor = te.compute(outer_shape, lambda i: x_tensor[i] + y_tensor[i], name="z")
-    s = te.create_schedule(z_tensor.op)
-
-    x_vtcm = s.cache_read(x_tensor, "global.vtcm", [z_tensor])
-    y_vtcm = s.cache_read(y_tensor, "global.vtcm", [z_tensor])
-    z_vtcm = s.cache_write(z_tensor, "global.vtcm")
-
-    layout_x_vtcm = s[x_vtcm].transform_layout(layout_transform_2d)
-    layout_y_vtcm = s[y_vtcm].transform_layout(layout_transform_2d)
-    _ = s[z_vtcm].transform_layout(layout_transform_2d)
-
-    mem_copy_read = intrin_mem_copy(inner_shape, dtype, "global.vtcm", "global")
-    s[x_vtcm].tensorize(layout_x_vtcm[1], mem_copy_read)
-    s[y_vtcm].tensorize(layout_y_vtcm[1], mem_copy_read)
-
-    # The loop schedule over `z` is not modified when calling `transform_layout`
-    # on `z_vtcm` above therefore we must call `split` to modify the loop schedule
-    # over `z` to match the layout of `z_vtcm` such that we can accurately write
-    # `z_vtcm` back to `z` using memory copy intrinsic
-    _, zinner = s[z_tensor].split(z_tensor.op.axis[0], factor=factor)
-    mem_copy_write = intrin_mem_copy(inner_shape, dtype, "global", "global.vtcm")
-    s[z_tensor].tensorize(zinner, mem_copy_write)
-
-    verify(hexagon_session, s, x_tensor, y_tensor, z_tensor, size)
-
-
-@T.prim_func
-def scale_by_two(buffer_a: T.Buffer[(8192,), "int8"], buffer_c: T.Buffer[(8192,), "int8"]):
-    for i in T.serial(
-        0,
-        8192,
-    ):
-        with T.block("C"):
-            buffer_c[i] = buffer_a[i] * T.int8(2)
-
-
-def test_vtcm_lowering():
-    """Test lowering with vtcm mem scope"""
-    mod = tvm.IRModule.from_expr(scale_by_two.with_attr("global_symbol", "main"))
-    sch = tir.Schedule(mod, debug_mask="all")
-    block_c = sch.get_block("C")
-    (flat,) = sch.get_loops(block_c)
-    outer, _, _, _ = sch.split(flat, factors=[8, 4, 2, 128])
-    cache_block = sch.cache_read(block_c, 0, storage_scope="global.vtcm")
-    sch.compute_at(cache_block, outer)
-    lowered = tvm.lower(sch.mod["main"])
-
-    def ir_module_has_allocate_nodes(irmod):
-        nallocs = 0
-
-        def _visit(stmt):
-            nonlocal nallocs
-            if isinstance(stmt, tvm.tir.Allocate):
-                nallocs += 1
-
-        tvm.tir.stmt_functor.post_order_visit(irmod["main"].body, _visit)
-        return nallocs
-
-    assert not ir_module_has_allocate_nodes(lowered), (
-        "AllocateNode found in lowered IRModule, "
-        "VTCM allocations should have been lowered to tir.nd_mem_alloc_with_scope"
-    )
```

Comment:
Let's keep this test as it is useful to check that no tir.Allocate nodes exist after lowering.